@types/jsb.d.ts

```diff
@@ -69,6 +69,25 @@ declare namespace jsb {
     export let onTouchEnd: TouchEventCallback | undefined;
     export let onTouchCancel: TouchEventCallback | undefined;
 
+    export interface ControllerInfo {
+        id: number;
+        axisInfoList: AxisInfo[],
+        buttonInfoList: ButtonInfo[],
+    }
+
+    export interface AxisInfo {
+        code: number,
+        value: number,
+    }
+
+    export interface ButtonInfo {
+        code: number,
+        isPressed: boolean,
+    }
+
+    export let onControllerInput: (infoList: ControllerInfo[]) => void | undefined;
+    export let onControllerChange: (controllerIds: number[]) => void | undefined;
```

Comment:
Is it need to be exported to developers?

Reply:
no, it's only for pal, 
The `Input.EventType.INPUT_CHANGE` is exported to developers